com.unity.ml-agents.extensions/Runtime/Sensors/ArticulationHierarchyUtil.cs

```diff
@@ -0,0 +1,60 @@
+#if UNITY_2020_1_OR_NEWER
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.MLAgents.Extensions.Sensors
+{
+
+    public class ArticulationHierarchyUtil : HierarchyUtil
+    {
+        ArticulationBody[] m_Bodies;
+
+        public ArticulationHierarchyUtil(ArticulationBody rootBody)
```

Comment:
ArticulationBodies are a little more constrained (pun intended) in how they can be set up - there's an explicit root body, and constrained bodies have to be lower in the hierarchy.